### ISSUE TITLE
`isolatedModules` documentation still says that it bans non-modules

### ISSUE THREAD

**Issue by DetachHead** [microsoft/TypeScript#55785]
### 🔎 Search Terms

isolatedmodules 5.0

### 🕗 Version & Regression Information

- This changed between versions 4.9.5 and 5.0.4

### ⏯ Playground Link

- [5.3.0-dev.20230919](https://www.typescriptlang.org/play?isolatedModules=true&ts=5.3.0-dev.20230919#code/MYewdgzgLgBAZiEMC8MCMQ) (no error)
- [4.9.5](https://www.typescriptlang.org/play?isolatedModules=true&ts=4.9.5#code/MYewdgzgLgBAZiEMC8MCMQ) (error)

### 💻 Code

```ts
const foo = 1
```

### 🙁 Actual behavior

no error

### 🙂 Expected behavior

```
'input.tsx' cannot be compiled under '--isolatedModules' because it is considered a global script file. Add an import, export, or an empty 'export {}' statement to make it a module.
```

### Additional information about the issue

i noticed there's a new `moduleDetection` compiler option. is `"moduleDetection": "force"` supposed to replace this use case of `isolatedModules`? if so, i would expect the error to still appear unless the file is detected as a module

**Comment by MartinJohns** [comment]
This is intentional. See #46295 and linked issues.

**Comment by DetachHead** [comment]
in that case, [the `isolatedModules` documentation](https://www.typescriptlang.org/tsconfig#non-module-files) needs to be updated, because it still says this:

> If `isolatedModules` is set, all implementation files must be modules (which means it has some form of `import`/`export`). An error occurs if any file isn’t a module

i always only used `isolatedModules` as a way to ban global scripts in my projects, i think many other people did the same, it's mentioned as a solution for this problem in many stackoverflow threads, eg:

https://stackoverflow.com/a/48360856
https://stackoverflow.com/a/56577324

so the updated documentation should probably also mention `"moduleDetection": false` as an alternative for this use case. seems like a pretty significant change that should have at least been mentioned in [the 5.0 announcement post](https://devblogs.microsoft.com/typescript/announcing-typescript-5-0/)
